components/camera/replaypcd/replaypcd_test.go

```diff
@@ -2,43 +2,397 @@ package replaypcd
+		test.That(t, err, test.ShouldBeNil)
+
+		startTime, err := time.Parse(timeFormat, replayCamCfg.Interval.Start)
+		test.That(t, err, test.ShouldBeNil)
+
+		// Iterate through files that meet the provided filter
+		for i := startTime.Second(); i < numPCDFiles; i++ {
+			pc, err := replayCamera.NextPointCloud(ctx)
+			test.That(t, err, test.ShouldBeNil)
+			test.That(t, pc, test.ShouldResemble, getPointCloudFromArtifact(t, i))
+		}
+
+		// Confirm the end of the dataset was reached when expected
+		pc, err := replayCamera.NextPointCloud(ctx)
+		test.That(t, err, test.ShouldNotBeNil)
+		test.That(t, err.Error(), test.ShouldContainSubstring, errEndOfDataset.Error())
+		test.That(t, pc, test.ShouldBeNil)
+
+		err = replayCamera.Close(ctx)
+		test.That(t, err, test.ShouldBeNil)
+
+		test.That(t, serverClose(), test.ShouldBeNil)
+	})
+
+	t.Run("Calling NextPointCloud with filter no data", func(t *testing.T) {
+		ctx := context.Background()
+
+		replayCamCfg := &Config{
+			Source: "test",
+			Interval: TimeInterval{
+				Start: "2000-01-01T12:00:30Z",
+				End:   "2000-01-01T12:00:40Z",
+			},
+		}
+
+		replayCamera, serverClose, err := createNewReplayPCDCamera(ctx, t, replayCamCfg, true)
+		test.That(t, err, test.ShouldBeNil)
+
+		// Confirm the end of the dataset was reached when expected
+		pc, err := replayCamera.NextPointCloud(ctx)
+		test.That(t, err, test.ShouldNotBeNil)
+		test.That(t, err.Error(), test.ShouldContainSubstring, errEndOfDataset.Error())
+		test.That(t, pc, test.ShouldBeNil)
+
+		err = replayCamera.Close(ctx)
+		test.That(t, err, test.ShouldBeNil)
+
+		test.That(t, serverClose(), test.ShouldBeNil)
+	})
+}
+
+func TestConfigValidation(t *testing.T) {
+	t.Run("Valid config with source and no timestamp", func(t *testing.T) {
+		replayCamCfg := &Config{
+			Source:   "test",
+			Interval: TimeInterval{},
+		}
+		deps, err := replayCamCfg.Validate("")
+		test.That(t, err, test.ShouldBeNil)
+		test.That(t, deps, test.ShouldResemble, []string{cloud.InternalServiceName.String()})
+	})
+
+	t.Run("Valid config with start timestamp", func(t *testing.T) {
+		replayCamCfg := &Config{
+			Source: "test",
+			Interval: TimeInterval{
+				Start: "2000-01-01T12:00:00Z",
+			},
+		}
+		deps, err := replayCamCfg.Validate("")
+		test.That(t, err, test.ShouldBeNil)
+		test.That(t, deps, test.ShouldResemble, []string{cloud.InternalServiceName.String()})
+	})
+
+	t.Run("Valid config with end timestamp", func(t *testing.T) {
+		replayCamCfg := &Config{
+			Source: "test",
+			Interval: TimeInterval{
+				End: "2000-01-01T12:00:00Z",
+			},
+		}
+		deps, err := replayCamCfg.Validate("")
+		test.That(t, err, test.ShouldBeNil)
+		test.That(t, deps, test.ShouldResemble, []string{cloud.InternalServiceName.String()})
+	})
+
+	t.Run("Valid config with start and end timestamps", func(t *testing.T) {
+		replayCamCfg := &Config{
+			Source: "test",
+			Interval: TimeInterval{
+				Start: "2000-01-01T12:00:00Z",
+				End:   "2000-01-01T12:00:01Z",
```

Comment:
robot_id is not checked in config validation as it is entirely optional and since the return on New() is a `camera.Camera` we don't have access to that parameter